### ISSUE TITLE
MbStringRule throws internal phpstan error when a function-to-avoid is detected

### ISSUE THREAD

**Issue by radonthetyrant** [Korbeil/phpstan-generic-rules#18]
In a file with "substr" instead of "mb_substr" the following error is printed out:

```
     Internal error: PHPStan\Analyser\RuleErrorTransformer::transform(): Argument #1 ($ruleError) must be of type PHPStan\Rules\RuleError, string        
     given, called in phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php on    
     line 145 while analysing file /home/radon/code/app/appapi/application/src/Infrastructure/TagBuilder.php                                   
     Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                           
     ## phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/RuleErrorTransformer.php(51)         
     #0 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(145):               
     PHPStan\Analyser\RuleErrorTransformer->transform()                                                                                                  
     #1 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassStatementsGatherer.php(116):        
     PHPStan\Analyser\FileAnalyser->{closure:PHPStan\Analyser\FileAnalyser::analyseFile():102}()                                                         
     #2 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(714):          
     PHPStan\Node\ClassStatementsGatherer->__invoke()                                                                                                    
     #3 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(818):          
     PHPStan\Analyser\NodeScopeResolver::{closure:PHPStan\Analyser\NodeScopeResolver::processStmtNode():713}()                                           
     #4 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(3505):         
     PHPStan\Analyser\NodeScopeResolver::{closure:PHPStan\Analyser\NodeScopeResolver::processStmtNode():817}()                                           
     #5 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(1989):         
     PHPStan\Analyser\NodeScopeResolver->callNodeCallbackWithExpression()                                                                                
     #6 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(4012):         
     PHPStan\Analyser\NodeScopeResolver->processExprNode()                                                                                               
     #7 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(2106):         
     PHPStan\Analyser\NodeScopeResolver->processArgs()                                                                                                   
     #8 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(2016):         
     PHPStan\Analyser\NodeScopeResolver->processExprNode()                                                                                               
     #9 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(4191):         
     PHPStan\Analyser\NodeScopeResolver->{closure:PHPStan\Analyser\NodeScopeResolver::processExprNode():2001}()                                          
     #10 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(2001):        
     PHPStan\Analyser\NodeScopeResolver->processAssignVar()                                                                                              
     #11 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(817):         
     PHPStan\Analyser\NodeScopeResolver->processExprNode()                                                                                               
     #12 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(520):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                                                               
     #13 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(713):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNodes()                                                                                              
     #14 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(520):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                                                               
     #15 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(897):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNodes()                                                                                              
     #16 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(520):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                                                               
     #17 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(848):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNodes()                                                                                              
     #18 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(468):         
     PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                                                               
     #19 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(203):              
     PHPStan\Analyser\NodeScopeResolver->processNodes()                                                                                                  
     #20 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(149):              
     PHPStan\Analyser\FileAnalyser->analyseFile()                                                                                                        
     #21                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.ph  
     p(111): PHPStan\Command\WorkerCommand::{closure:PHPStan\Command\WorkerCommand::runWorker():126}()                                                   
     #22 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/clue/ndjson-react/src/Decoder.php(117):   
     _PHPStan_6597ef616\Evenement\EventEmitter->emit()                                                                                                   
     #23                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.ph  
     p(111): _PHPStan_6597ef616\Clue\React\NDJson\Decoder->handleData()                                                                                  
     #24 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/Util.php(62):            
     _PHPStan_6597ef616\Evenement\EventEmitter->emit()                                                                                                   
     #25                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.ph  
     p(111): _PHPStan_6597ef616\React\Stream\Util::{closure:_PHPStan_6597ef616\React\Stream\Util::forwardEvents():61}()                                  
     #26                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/DuplexResourceStream.php(16  
     8): _PHPStan_6597ef616\Evenement\EventEmitter->emit()                                                                                               
     #27                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(20  
     1): _PHPStan_6597ef616\React\Stream\DuplexResourceStream->handleData()                                                                              
     #28                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(17  
     3): _PHPStan_6597ef616\React\EventLoop\StreamSelectLoop->waitForStreamActivity()                                                                    
     #29 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(105):              
     _PHPStan_6597ef616\React\EventLoop\StreamSelectLoop->run()                                                                                          
     #30                                                                                                                                                 
     phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Command/Command.php(259):     
     PHPStan\Command\WorkerCommand->execute()                                                                                                            
     #31 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(868):     
     _PHPStan_6597ef616\Symfony\Component\Console\Command\Command->run()                                                                                 
     #32 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(261):     
     _PHPStan_6597ef616\Symfony\Component\Console\Application->doRunCommand()                                                                            
     #33 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(157):     
     _PHPStan_6597ef616\Symfony\Component\Console\Application->doRun()                                                                                   
     #34 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(96):                                 
     _PHPStan_6597ef616\Symfony\Component\Console\Application->run()                                                                                     
     #35 phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(97):                                 
     {closure:phar:///home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan:14}()                           
     #36 /home/radon/code/app/appapi/tools/phpstan/vendor/phpstan/phpstan/phpstan.phar(7): require('...')                                      
     #37 /home/radon/code/app/appapi/tools/phpstan/vendor/bin/phpstan.phar(119): include('...')                                                
     #38 {main}
```

composer.json
```
{
  "require": {
    "php": ">=8.4",
    "phpstan/phpstan": "^2.1",
    "phpstan/extension-installer": "^1.4",
    "phpstan/phpstan-deprecation-rules": "^2.0",
    "jetbrains/phpstorm-attributes": "^1.2",
    "korbeil/phpstan-generic-rules": "^2.1"
  },
  "config": {
    "allow-plugins": {
      "phpstan/extension-installer": true
    }
  }
}
```

To reproduce: Run with versions mentioned above, create a file with substr() and that should produce the error.
Replace "substr" with "mb_substr" and the internal error goes away
